Temporarily deploy webhooks with failurePolicy set to Ignore

mmirecki opened the pull request:
Deploying webhooks with a non Ignore failure policy can negatively affect the stability of the cluster if the deployment backing it fails. This PR causes the webhooks to be created with an Ignore failure policy, and changes it to the desired value once the gatekeeper controller deployment is fully operational

font commented on controllers/gatekeeper_controller.go:
```diff
@@ -194,69 +199,137 @@ func (r *GatekeeperReconciler) SetupWithManager(mgr ctrl.Manager) error {
-			continue
-		case a == RoleFile && r.isOpenShift():
-			a = openshiftAssetsDir + a
+	// Checking for deployment before deploying assets to avoid cert rotator erors
```
s/erors/errors/

mmirecki replied:
Done

font commented on controllers/merge/merge.go:
```diff
@@ -39,6 +39,8 @@ func RetainClusterObjectFields(desiredObj, clusterObj *unstructured.Unstructured
 		fallthrough
 	case util.MutatingWebhookConfigurationKind:
 		return retainWebhookConfigurationFields(desiredObj, clusterObj)
+	case util.SecretKind:
```
Doh, good catch!

font replied:
@mmirecki I'm actually thinking that this change may be all that was needed to make #149 work successfully. WDYT?

font commented on test/e2e/gatekeeper_controller.go:
```diff
@@ -46,9 +46,9 @@ import (
 const (
 	// The length of time between polls.
-	pollInterval = 50 * time.Millisecond
+	pollInterval = time.Second
```
This is OK for now, but I'd prefer to leave a low default for local testing and allow it to be configurable via command line so we can set a higher value when running in CI.

mmirecki replied:
Done

font commented on test/e2e/gatekeeper_controller.go:
```diff
@@ -46,9 +46,9 @@ import (
+	pollInterval = time.Second
 	// How long to try before giving up.
-	waitTimeout = 30 * time.Second
+	waitTimeout = 60 * time.Second
```
Same comment as above.

mmirecki replied:
Done

font reviewed (commented):
Amazing @mmirecki! I made a couple comments and one minor nit. I re-ran the tests to verify it passes consistently, but ran into an issue that I've fixed in #151. Please rebase so the test passes again.

mmirecki commented:
> Amazing @mmirecki! I made a couple comments and one minor nit. I re-ran the tests to verify it passes consistently, but ran into an issue that I've fixed in #151. Please rebase so the test passes again.

Rebased
